src/commands/plug.rs

```diff
@@ -53,24 +151,66 @@ impl PlugCommand {
         // Collect the plugs by their names
         let mut plugs_by_name = std::collections::HashMap::<_, Vec<_>>::new();
         for plug in self.plugs.iter() {
-            let name = plug
-                .file_stem()
-                .map(|fs| fs.to_string_lossy())
-                .with_context(|| format!("path to plug '{}' was not a file", plug.display()))?;
+            let name = match plug {
+                #[cfg(feature = "registry")]
+                PackageRef::RegistryPackage((name, _)) => std::borrow::Cow::Borrowed(name.as_ref()),
+                PackageRef::LocalPath(path) => path
+                    .file_stem()
+                    .map(|fs| fs.to_string_lossy())
+                    .with_context(|| format!("path to plug '{}' was not a file", plug))?,
+            };
+
             // TODO(rylev): sanitize the name to ensure it's a valid package identifier.
             plugs_by_name.entry(name).or_default().push(plug);
         }
 
         // Plug each plug into the socket.
-        for (name, plug_paths) in plugs_by_name {
-            for (i, plug_path) in plug_paths.iter().enumerate() {
-                let mut name = format!("plug:{name}");
+        for (name, plug_refs) in plugs_by_name {
+            for (i, plug_ref) in plug_refs.iter().enumerate() {
+                let (mut name, path) = match plug_ref {
+                    #[cfg(feature = "registry")]
+                    PackageRef::RegistryPackage((name, version)) => {
+                        let client = client.as_ref().ok_or_else(|| {
+                            anyhow::anyhow!(
+                                "Warg registry is not configured. Package `{name}` was not found."
+                            )
+                        })?;
+
+                        let path = if let Some(ver) = version {
+                            let download = client.download_exact(name, ver).await?;
+                            println!(
```

Comment:
We're not currently printing anything else when running the plug command. I think all of these `println!`s should be `log::debug` statements instead. In a follow up, we can figure out which output to always show the user.

Reply:
Fixed.